stdlib/prelude.an

```diff
@@ -263,8 +263,8 @@ impl TryCast u64 i64 with
         if x > 9_223_372_036_854_775_807 then None
         else Some (transmute x)
 
-trait Append t with
-    (++): t - t -> t
+trait Append a b with
+    (++): a - b -> a
```

Comment:
What is the reasoning on this change?
Requiring both operands to be the same type makes it a Semigroup and was a deliberate decision to aid type inference, which makes impls easier to find. I think we should keep `Append t` until we have a good usecase for a mixed append at least. Potentially `Append StringBuilder string` in the future, but we may also want to generalize the return type as well.